haskell/src/SwiftNav/SBP/TH.hs

```diff
@@ -0,0 +1,30 @@
+-- |
+-- Module:      SwiftNav.SBP.TH
+-- Copyright:   Copyright (C) 2015 Swift Navigation, Inc.
+-- License:     LGPL-3
+-- Maintainer:  Mark Fine <dev@swiftnav.com>
+-- Stability:   experimental
+-- Portability: portable
+--
+-- Templated generation of SBP interfaces.
+
+module SwiftNav.SBP.TH where
+
+import BasicPrelude
+import Data.Binary
+import Data.ByteString
+import Data.ByteString.Lazy hiding ( ByteString )
+import Data.Word
+import Language.Haskell.TH
+import SwiftNav.SBP.Types
+
+-- | Derive ToSBP typeclass, given an SBP message type name and the
+-- name of the implemented type.
+deriveSBP :: Name -> Name -> Q [Dec]
+deriveSBP msgType name = [d|instance ToSBP $(conT name) where
+                              toSBP msg senderID = encoded { _msgSBPCrc = checkCrc encoded }
```

Comment:
@mfine so I tried lens'ing here and got some error that seemed like it shouldn't be the case: 

```
src/SwiftNav/SBP/Acquisition.hs:66:3:
    Couldn't match expected type ‘Msg’ with actual type ‘s0 -> t0’
    In the expression:
      (encoded_a7u6 .~ (msgSBPCrc (checkCrc encoded_a7u6)))
    In an equation for ‘toSBP’:
        toSBP msg_a7u2 senderID_a7u3
          = (encoded_a7u6 .~ (msgSBPCrc (checkCrc encoded_a7u6)))
          where
              payload_a7u4 = (toStrict $ (encode msg_a7u2))
              len_a7u5 = (fromIntegral $ (Data.ByteString.length payload_a7u4))
              encoded_a7u6
                = Msg msgAcqResult senderID_a7u3 len_a7u5 payload_a7u4 0
```